Client/Application/AppDelegate.swift

```diff
@@ -851,30 +853,29 @@ class AppSyncDelegate: SyncDelegate {
 
         // check to see what the current notification settings are and only try and send a notification if
         // the user has agreed to them
-        if let currentSettings = app.currentUserNotificationSettings {
-            if currentSettings.types.rawValue & UIUserNotificationType.alert.rawValue != 0 {
+        UNUserNotificationCenter.current.getNotificationSetings { settings in
+            if settings.alertSetting == UNNotificationSetting.enabled {
                 if Logger.logPII {
                     log.info("Displaying notification for URL \(url.absoluteString)")
                 }
 
-                let notification = UILocalNotification()
-                notification.fireDate = Date()
-                notification.timeZone = NSTimeZone.default
+                let notificationContent = UNMutableNotificationContent()
                 let title: String
                 if let deviceName = deviceName {
                     title = String(format: Strings.SentTab_TabArrivingNotification_WithDevice_title, deviceName)
                 } else {
                     title = Strings.SentTab_TabArrivingNotification_NoDevice_title
                 }
-                notification.alertTitle = title
-                notification.alertBody = url.absoluteDisplayString
-                notification.userInfo = [TabSendURLKey: url.absoluteString, TabSendTitleKey: title]
-                notification.alertAction = nil
+                notificationContent.title = title
+                notificationContent.body = url.absoluteDisplayString
+                notificationContent.userInfo = [TabSendURLKey: url.absoluteString, TabSendTitleKey: title]
+                notificationContent.categoryIdentifier = TabSendCategory
+
+                let trigger = UNTimeIntervalNotificationTrigger(timeInterval: 1, repeats: false)
 
-                // Restore this when we fix Bug 1364420.
-                // notification.category = TabSendCategory
```

Comment:
[Bug 1364420][1] is about adding categories to push notifications from sent tabs: sent tabs come via sync commands, and the push notification is essentially a sync tickle which catches the send tab sync command. 

This change doesn't seem to add actions to the push notifications (see `NotificationService` and `FxAPushMessageHandler`), however, you may have identified a route to it. 

[1]: https://bugzilla.mozilla.org/show_bug.cgi?id=1364420